src/wallet/wallet.cpp

```diff
@@ -1818,32 +1818,37 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
     return result;
 }
 
+std::vector<CWalletTx*> CWallet::GetSortedTxs()
+{
+    AssertLockHeld(cs_wallet);
+    // Sort pending transactions based on their initial wallet insertion order
+    std::map<int64_t, CWalletTx*> sorted_txs;
+    for (auto& [wtxid, wtx] : mapWallet) {
+        assert(wtx.GetHash() == wtxid);
+        sorted_txs.insert(std::make_pair(wtx.nOrderPos, &wtx));
+    }
+    std::vector<CWalletTx*> out;
+    for (auto [_, wtx] : sorted_txs) {
+        out.push_back(wtx);
+    }
+    return out;
+}
+
 void CWallet::ReacceptWalletTransactions()
 {
     // If transactions aren't being broadcasted, don't let them into local mempool either
     if (!fBroadcastTransactions)
         return;
-    std::map<int64_t, CWalletTx*> mapSorted;
 
-    // Sort pending wallet transactions based on their initial wallet insertion order
-    for (std::pair<const uint256, CWalletTx>& item : mapWallet) {
-        const uint256& wtxid = item.first;
-        CWalletTx& wtx = item.second;
-        assert(wtx.GetHash() == wtxid);
-
-        int nDepth = GetTxDepthInMainChain(wtx);
+    // Try to add wallet transactions to memory pool
+    for (auto wtx : GetSortedTxs()) {
+        int nDepth = GetTxDepthInMainChain(*wtx);
 
-        if (!wtx.IsCoinBase() && (nDepth == 0 && !wtx.isAbandoned())) {
-            mapSorted.insert(std::make_pair(wtx.nOrderPos, &wtx));
+        if (!wtx->IsCoinBase() && (nDepth == 0 && !wtx->isAbandoned())) {
```

Comment:
When looking at `CWallet::GetTxDepthInMainChain`, if `nDepth == 0`, isn't that equivalent to `!wtx->IsConfirmed() && !wtx->IsConflicted()`, which in turn in combination with `!wtx->isAbandoned()` is equivalent to `!wtx->IsUnconfirmed()` - am I missing something? (I know this is moved code)
```suggestion
        if (!wtx->IsCoinBase() && !wtx->IsUnconfirmed()) {
```

Reply:
Yes, those should be equivalent. Will do if I need to re-push.

Reply:
Done